src/tribler/core/components/popularity/community/popularity_community.py

```diff
@@ -35,91 +37,38 @@ def __init__(self, *args, torrent_checker=None, **kwargs):
         self.torrent_checker = torrent_checker
 
         self.add_message_handler(TorrentsHealthPayload, self.on_torrents_health)
+        self.add_message_handler(PopularTorrentsRequest, self.on_popular_torrents_request)
 
-        self.logger.info('Popularity Community initialized (peer mid %s)',
-                         hexlify(self.my_peer.mid))
-        self.register_task("gossip_popular_torrents", self.gossip_popular_torrents_health,
-                           interval=PopularityCommunity.GOSSIP_INTERVAL_FOR_POPULAR_TORRENTS)
+        self.logger.info('Popularity Community initialized (peer mid %s)', hexlify(self.my_peer.mid))
         self.register_task("gossip_random_torrents", self.gossip_random_torrents_health,
                            interval=PopularityCommunity.GOSSIP_INTERVAL_FOR_RANDOM_TORRENTS)
 
         # Init version community message handlers
         self.init_version_community()
 
-    @staticmethod
-    def select_torrents_to_gossip(torrents, include_popular=True, include_random=True) -> (set, set):
-        """ Select torrents to gossip.
+    def introduction_request_callback(self, peer, _dist, _payload):
```

Comment:
This overloading of the `introduction_request_callback` breaks the inheritance chain of the method which could lead to really strange and hard-to-detect bugs.

```suggestion
    def introduction_request_callback(self, peer, dist, payload):
        super().introduction_request_callback(peer, dist, payload)
```

Reply:
Good point, thank you for the reminder.